processor.go

```diff
@@ -169,14 +169,22 @@ func (p *processor) handle(ctx *fh.RequestCtx) {
 		return
 	}
 
+	tenantPrefix := p.cfg.Tenant.Prefix
+
+	if p.cfg.Tenant.PrefixPreferSource {
+		if string(ctx.Request.Header.Peek("X-Scope-OrgID")) != "" {
```

Comment:
1. Maybe we should peek only once, save to a string and then use that
2. `X-Scope_OrgID` probably should be replaces with `config.Tenant.Header`